src/main/java/com/solr/clientwrapper/rest/SolrSearchOperationResource.java

```diff
@@ -0,0 +1,115 @@
+package com.solr.clientwrapper.rest;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.SolrQuery.SortClause;
+import org.apache.solr.client.solrj.beans.DocumentObjectBinder;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocumentList;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.solr.clientwrapper.infrastructure.solrbean.SolrCollectionIndex;
+import com.solr.clientwrapper.infrastructure.solrbean.SolrSearchResult;
+
+@RestController
+@RequestMapping("/solr")
+public class SolrSearchOperationResource {
+	/*
+	 * /////// ############## Solr Search Operation ################ /////////
+	 * Author: Piyush Ojha; NeoSOFT Tech.
+	 */    
+	
+    private static final String SOLR_DATA_NAME = "techproducts";
```

Comment:
this should be taken in request params

Reply:
Yeah, I am taking it from RequestParams. I will just remove this constant assignment, as it will not be required.

Reply:
Done.